CONTRIBUTING.md

```diff
@@ -1,3 +1,32 @@
-# Contributing
+# Contributing to Noe Core
 
-## Pull request workflow
+Thank you for considering contribution to Noe Core. As a potential contributor, your changes and ideas are always welcome. Please do not ever hesitate to ask a question or send a Pull Request.
+
+## How Can I Contribute?
+
+### Reporting Bugs
+If you come across what seems to be a bug, please use the Github [issue tracker](https://github.com/web-servers/noe-core/issues) and file an issue. An issue can also be used for discussions. 
+
+### Contributing Documentation
+Documentation is crucial for success of any project. If you feel like there is lack of documentation in some area, feel free to [file an issue](https://github.com/web-servers/noe-core/issues). If you want to contribute documentation, feel free to open a PR with your changes. Your contribution may or may not be changed with stylistic and grammatical corrections; do not expect to see your changes ad verbum in the code.
+
+### Contributing Code
+To limit wasted effort and time, before contributing code, please open up an issue where you describe your desired change in Noe Core. After a debate and review of the idea, follow the Contribution Workflow section to get your patch accepted.
+
+## Contribution Workflow
+Noe Core uses the standard Github's [Pull Request](https://help.github.com/articles/about-pull-requests/) workflow. In addition, we expect you to:
+
+1. Adhere to the review provided within the pull request. If you disagree, we are always open to a debate.
+1. Adhere to the Apache software Foundation's [Code of Conduct](http://www.apache.org/foundation/policies/conduct.html).
+1. Provide integration and unit test where applicable.
+1. Provide full documentation for your changes where applicable.
+1. Provide reasonable commit messages.
+1. Adhere to our codestyle. 
+
+We use the same coding style as [Wildfly](https://github.com/wildfly/wildfly-core/tree/master/ide-configs) with the following exceptions:
+ - Indentation size is 2 spaces.
+ - Contined indentation is 4 spaces.
```

Comment:
@m-czernek, 
Hmm, ad Indentation/Contined indentation -- do we though? I don't want us to be influenced by the legacy Groovy style. I would like us to follow Sunstone in the style and possibly also release config etc.  They use Wildfly style without modifications IIUC: [see the pom](https://github.com/wildfly-extras/sunstone/blob/master/pom.xml#L60).

Reply:
We definitely do follow 2 spaces tab - 4 spaces continuation at this moment...  Do we want to re-format Noe Core? Seems like [Wildfly](https://github.com/wildfly/wildfly-core/blob/master/ide-configs/eclipse/wildfly-formatter.xml#L43) indeed is different in the continued indentation.